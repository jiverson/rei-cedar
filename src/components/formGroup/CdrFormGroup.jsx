import clsx from 'clsx';
import style from './styles/CdrFormGroup.scss';
import IconErrorStroke from '../icon/comps/error-stroke';

export default {
  name: 'CdrFormGroup',
  components: {
    IconErrorStroke,
  },
  props: {
    label: {
      type: String,
      default: '',
      required: false,
    },
    error: Boolean,
    required: Boolean,
    optional: Boolean,
  },
  data() {
    return {
      style,
    };
  },
  computed: {
    baseClass() {
      return 'cdr-form-group';
    },
    errorClass() {
      return this.error ? this.style['cdr-form-group--error'] : '';
    },
  },
  render() {
    return (<fieldset class={this.style[this.baseClass]}>
      <legend>
        {this.$slots.label || this.label}
        {this.required && (
          <span
            class={this.style['cdr-form-group__required']}
            aria-label="required"
          >
            *
          </span>
        )}

        {this.optional && (
          <span
            class={this.style['cdr-form-group__optional']}
          >
            (optional)
          </span>
        )}
      </legend>
      <div class={clsx(this.style['cdr-form-group__wrapper'], this.errorClass)}>
        {this.$slots.default}
      </div>
      {this.error && this.$slots.error && (
        <div class={this.style['cdr-form-group__error-message']}>
          <icon-error-stroke inherit-color/> { this.$slots.error }
        </div>
      )}
    </fieldset>);
  },
};
