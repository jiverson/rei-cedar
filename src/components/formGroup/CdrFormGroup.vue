<template>
<fieldset
  class={clsx(this.style[this.baseClass], this.disabledClass)}
  disabled={this.disabled}
>
  <legend>
    {this.$slots.label || this.label}
    {requiredEl || optionalEl ? ' ' : ''}
    {requiredEl || optionalEl}



      const requiredEl = this.required ? (
        <span aria-label="required">
          *
        </span>
      ) : '';

      const optionalEl = this.optional ? (
        <span
          class={this.style['cdr-form-group__optional']}
        >
          (optional)
        </span>
      ) : '';




  </legend>
  <div class={clsx(this.style['cdr-form-group__wrapper'], this.errorClass)}>
    {this.$slots.default}
  </div>
  {this.error && (
    <cdr-form-error error={this.error}>
      <template slot="error">
        {this.$slots.error}
      </template>
    </cdr-form-error>
  )}
</fieldset>
</template>
<script lang="ts">
import { defineComponent, computed } from 'vue';
import clsx from 'clsx';
import style from './styles/CdrFormGroup.scss';
import IconErrorStroke from '../icon/comps/error-stroke';
import CdrFormError from '../formError/CdrFormError';
import { buildClass } from '../../utils/buildClass'
import propValidator from '../../utils/propValidator';



export default {
  name: 'CdrFormGroup',
  components: {
    IconErrorStroke,
    CdrFormError,
  },
  props: {
    label: {
      type: String,
      default: '',
      required: false,
    },
    // Set error styling
    error: {
      type: [Boolean, String],
      default: false,
    },
    required: Boolean,
    optional: Boolean,
    disabled: Boolean,
  },
  setup(props) {
    // baseClass() {
    //   return 'cdr-form-group';
    // },
    // errorClass() {
    //   return this.error ? this.style['cdr-form-group--error'] : '';
    // },
    // disabledClass() {
    //   return this.disabled ? this.style['cdr-form-group--disabled'] : '';
    // },
    return {};
  },
};
</script>
