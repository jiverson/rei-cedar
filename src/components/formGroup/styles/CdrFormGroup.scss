@import '../../../css/settings/index.scss';
@import './CdrFormGroup.vars.scss';

.cdr-form-group {
  @include cdr-form-group-base-mixin;
}
