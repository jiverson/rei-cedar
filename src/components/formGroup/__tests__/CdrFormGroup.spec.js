import { mount } from '@vue/test-utils';
import CdrFormGroup from 'componentdir/formGroup/CdrFormGroup';

describe('CdrFormGroup', () => {
  test('renders correctly', () => {
    const wrapper = mount(CdrFormGroup);
    expect(wrapper.element).toMatchSnapshot();
  });
});
