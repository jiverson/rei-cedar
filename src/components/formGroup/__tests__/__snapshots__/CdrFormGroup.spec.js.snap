// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CdrFormGroup renders correctly 1`] = `
<fieldset
  class="cdr-form-group"
>
  <legend />
</fieldset>
`;
